### PR TITLE
feat(terminal): resume AI sessions after relaunch

### PR DESCRIPTION
**Pull request by dca**
## Summary

- persist each local terminal pane exact Claude Code or Codex session identity
- automatically resume every persisted AI pane after an update, crash, or relaunch
- add an independent conversation recovery setting while keeping terminal history restoration unchanged
- extend the status-hook protocol with an explicit agent field and retain compatibility with older hook messages

## Behavior

- resumes with the existing safe commands: `claude --resume <id>` and `codex resume <id>`
- eagerly mounts inactive tabs that contain conversations requiring recovery
- clears a binding when its matching session ends or the pane is closed
- ignores SSH panes, malformed session IDs, duplicate remount attempts, and delayed SessionEnd events from replaced sessions
- keeps status hooks installed while either visible status tracking or conversation recovery needs them

Interrupted tool operations are not rerun; the CLI resumes the exact conversation context.

## Testing

- `pnpm test` — 214 files, 1879 tests passed
- `pnpm typecheck`
- `pnpm build`
- `cargo +1.89.0 test --manifest-path src-tauri/Cargo.toml` — 447 passed, 6 ignored
- `cargo +1.89.0 check --manifest-path src-tauri/Cargo.toml`
- `git diff --check`

### CODE REVIEW

File: src/modules/terminal/TerminalView.tsx [hunk 1]
```diff
@@ -1043,6 +1105,9 @@ export function TerminalView({
 
     return () => {
       disposed = true;
+      if (autoResumeCommand && !autoResumeStarted) {
+        releaseAutoResumeAttempt(initialLeafId, initialAiSession);
+      }
```
[the remark this answers is not in the document]
Comment (dca) [reply src/modules/terminal/TerminalView.tsx]: Fixed in 4cdc37a. The resume claim is now released unconditionally whenever the owning pane unmounts, and the no-longer-needed autoResumeStarted state was removed. This lets StrictMode or layout remounts claim and resume the conversation in the replacement PTY. Verified with TypeScript typecheck and 121 focused regression tests.

**Review by mukiwu — COMMENTED** [review]
Thanks for the PR — the wire-protocol extension is backward-compatible, the resume command builder is reused with its shell-safety guard intact, and the test coverage is thorough. Three points before this can move forward; the first is the important one.

**1. `autoResumeAiSessions` defaulting to `true` re-installs hooks for users who explicitly opted out**

`App.tsx` now installs the hooks when `claudeStatusTracking || autoResumeAiSessions`. Uninstalling the hooks when tracking is toggled off is a deliberate design decision in this app: writing into `~/.claude/settings.json` and the Codex hook config is invasive, and the uninstall path exists precisely because we don't want to touch the agent config of users who opted out. Some users turned tracking off for exactly that reason — and with the new setting defaulting to `true`, they upgrade and get the hooks silently re-installed, plus the app starts typing resume commands into their shells on relaunch.

Please default the new toggle to `false` (opt-in). Related: unchecking "Track Claude session status" now silently keeps the hooks installed while recovery is on — worth a hint in the recovery toggle's description so users know it also keeps the hooks.

**2. Recovery also triggers after a normal quit, not just update/crash**

On a graceful app quit the CLI's SessionEnd ping most likely never reaches the loopback listener (it goes down with the app), so the binding survives and the next launch resumes unconditionally. That may well be the desired semantics, but the settings copy says "After an update, relaunch, or unexpected exit…", which reads as if resuming is scoped to those cases. Behavior and wording should match — if resume-on-any-relaunch is intended, please adjust the description.

**3. `claude --resume <id>` implicitly depends on the restored cwd**

Claude Code resolves sessions per project directory, so resume only works when the pane's restored cwd still matches the original one. If that directory was deleted or renamed, the user sees a failing command in the pane on launch. Probably an acceptable edge case, but worth noting (or handling gracefully), since the persisted binding can outlive the directory.

**Comment by dca** [comment]
Addressed the latest review in 4a077c8:

- Changed `autoResumeAiSessions` to default to `false`, so recovery is opt-in and existing users who disabled status tracking will not have hooks silently reinstalled.
- Updated the setting copy to accurately say recovery runs whenever tempo-term starts, including ordinary quit/relaunch.
- Clarified that enabling recovery keeps the Claude/Codex status hooks installed to remember session IDs.
- Documented the accepted cwd edge case: resume may fail if the original project directory no longer exists.

Validation: full frontend suite (214 files / 1879 tests), `pnpm typecheck`, `pnpm build`, and `git diff --check` all pass.

**Review by mukiwu — APPROVED** [review]
All three review points addressed cleanly, and the unmount-release fix covers the StrictMode double-mount case. Thanks for the quick turnaround!
